### PR TITLE
[Merged by Bors] - feat(Data.Real.ENatENNReal/EReal): add a few coe lemmas

### PR DESCRIPTION
**Pull request by D-Thomine**
Add a few lemmas about coercion:

- `toENNReal_coe_eq_iff`, `toENNReal_pow` in Data.Real.ENatENNReal
- 9 lemmas about `natCast` into `EReal ` in Data.Real.EReal

Allows a better handle on functions from ENat to ENNReal, as well as from Nat to EReal.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by j-loreaux — COMMENTED** [review]
Thanks, these seem helpful.

**Comment by j-loreaux** [comment]
!bench

File: Mathlib/Data/Real/EReal.lean [hunk 1]
```diff
@@ -728,6 +728,33 @@ theorem coe_ennreal_nsmul (n : ℕ) (x : ℝ≥0∞) : (↑(n • x) : EReal) =
...
+
+@[simp, norm_cast]
+theorem natCast_le_iff {m n : ℕ} : (m : EReal) ≤ (n : EReal) ↔ m ≤ n := by
+  rw [← coe_coe_eq_natCast n, ← coe_coe_eq_natCast m, EReal.coe_le_coe_iff, Nat.cast_le]
+
+@[simp, norm_cast]
+theorem natCast_lt_iff {m n : ℕ} : (m : EReal) < (n : EReal) ↔ m < n := by
+  rw [← coe_coe_eq_natCast n, ← coe_coe_eq_natCast m, EReal.coe_lt_coe_iff, Nat.cast_lt]
+
+theorem natCast_mul (m n : ℕ) :
```
[the remark this answers is not in the document]
Comment (D-Thomine) [reply Mathlib/Data/Real/EReal.lean:755]: No real reason, I just forgot.

By the way, I am a bit unsure about when to put the `norm_cast` attribute; the library seems inconsistent about it. Should I put it on e.g. `toENNReal_pow`?

[the remark this answers is not in the document]
Comment (j-loreaux) [reply Mathlib/Data/Real/EReal.lean:755]: Yes, please do. There are a few rules for which things should be `norm_cast`, but the attribute will often tell you if it's wrong to do it.

[the remark this answers is not in the document]
Comment (D-Thomine) [reply Mathlib/Data/Real/EReal.lean:755]: Done. I've also added the `norm-cast` attribute to a bunch of other lemmas about `toENNReal` (there was no reason for `toENNReal_zero` to have this attribute and not `toENNReal_one`, or something like `EReal.coe_mul` but not `toENNReal_mul`).

**Comment by j-loreaux** [comment]
Great thanks!

bors merge

**Pull request closed without merging**